src/app/shared/state/extended-workflow.service.ts

```diff
@@ -49,6 +51,18 @@ export class ExtendedWorkflowService {
     }
   }
 
+  /**
+   * Updates the extendedWorkflow if has the same id as workflow
+   *
+   * @param workflow
+   */
+  updateIfActive(workflow: Workflow) {
+    const extendedWorkflow = this.extendedWorkflowQuery.getValue();
+    if (extendedWorkflow && extendedWorkflow.id === workflow.id) {
+      this.update(workflow);
```

Comment:
You're updating the extendedworkflow store but not the original workflow in the normal workflowstore.  They're going to be out of sync.

Reply:
> You're updating the extendedworkflow store but not the original workflow in the normal workflowstore. They're going to be out of sync.

This is true, but it's how I interpreted [your comment here](https://github.com/dockstore/dockstore-ui2/pull/987#discussion_r431871770) that it wasn't necessary. I read into it thinking your point was that when you switch workflows, the UI fetches the latest anyways.

My original commit was updating both stores, but then you had that comment, so I'm not clear exactly what you want. Although in hindsight maybe your comment was just a question and not a request for change.

Are you saying:

1. If it's the active workflow, update the workflowstore and the extendedworkflow store; don't update the workflow store if it's not the active workflow?
2. Or, just update both stores?

Reply:
Basically #1

I was expecting a check for active and then `workflowService.setWorkflow()` if it is

Reply:
> I was expecting a check for active and then `workflowService.setWorkflow()` if it is

Why not update the workflowstore if it isn't active? It can become active.

Reply:
because if you swap to another workflow, it's going to trigger a more complete GET request, with validations and aliases.

Reply:
> because if you swap to another workflow, it's going to trigger a more complete GET request, with validations and aliases. 

You approved, but as justification, I feel like the store might as well be up to date; there might be some flow some day that relies on it.

> Uhoh, should the workflow be using the response for any workflow then?

I don't understand; can you elaborate?

Reply:
" it's going to trigger a more complete GET request, with validations and aliases."

The current responses you're getting are incomplete.